### PR TITLE
WIP Add Docusaurus site build artifacts

### PR DESCRIPTION
**Pull request by elmiko**
This change adds files and artifacts to produce a [Docusaurus](https://docusaurus.io/) based web site from the files in the `docs` directory. It also adds some readme files to help instruct about adding more content, header material to current markdown files, and a make target to build a containerized version of the site.

### CODE REVIEW

**Comment by Danil-Grigorev** [comment]
Tried it out locally, works like a charm. Thanks @elmiko :D
image

/lgtm

**Review by Danil-Grigorev — APPROVED** [review]

File: docusaurus/.gitignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.DS_Store
+
+node_modules
+
+lib/core/metadata.js
+lib/core/MetadataBlog.js
+
+website/translated_docs
+website/build/
+website/yarn.lock
```
Comment (JoelSpeed) [inline docusaurus/.gitignore:10]: Do we want to ignore the yarn.lock? Isn't that the node equivalent of ignoring the go.sum?

File: docusaurus/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+in the [Docusaurus documentation](https://docusaurus.io/docs/en/installation).
+
+## Building a local version of the site
+
+The easiest way to build the site locally is using a container runtime like
+[Podman](https://podman.io). Use the command `make docusaurus-image` to build
+a local container image named `machine-api-operator-book:latest`.
+
+If you wish to customize the build, please see the `docusaurus/Dockerfile` file
+for more information.
```
Comment (JoelSpeed) [inline docusaurus/README.md:31]: Maybe we should add a docker example command of how to run this, or a make target for it, something along the lines of (unsure if correct port):
```
docker run --rm -it -p 3000:3000 machine-api-docusaurus 
```

File: docusaurus/website/siteConfig.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  /* path to images for header/footer */
+  /*
+  headerIcon: 'img/favicon.ico',
+  footerIcon: 'img/favicon.ico',
+  favicon: 'img/favicon.ico',
+  */
+
+  /* Colors for website */
+  colors: {
+    primaryColor: '#000000',
```
Comment (JoelSpeed) [inline docusaurus/website/siteConfig.js:40]: That's a boring primary 😉

**Review by JoelSpeed — COMMENTED** [review]
Looks great, have a couple of suggestions but nothing major, happy to give the approval if you don't want to take my suggestions on board

File: docusaurus/.gitignore (hunk 1 above)
Comment (elmiko) [reply docusaurus/.gitignore:10]: good spot, that was the default gitignore from docusaurus. we should definitely keep the lock file.

File: docusaurus/README.md (hunk 2 above)
Comment (elmiko) [reply docusaurus/README.md:31]: yeah, i was thinking about that after i pushed the commit XD

File: docusaurus/website/siteConfig.js (hunk 3 above)
Comment (elmiko) [reply docusaurus/website/siteConfig.js:40]: my original thought was to use a palette derived from colors at brand.redhat.com, but i focused more on content that style ;)

Comment (Danil-Grigorev) [reply docusaurus/website/siteConfig.js:40]: It is very minimalistic and I like it personally :)

**Comment by elmiko** [comment]
updated:
* section to docusaurus readme on running the local container
* remove yarn.lock from docusaurus gitignore
* add an initial yarn.lock

**Comment by elmiko** [comment]
marking this as wip to reduce the tests and also to give us a chance to discuss if we want to continue with this.

/retitle WIP Add Docusaurus site build artifacts

**Comment by elmiko** [comment]
@JoelSpeed should we close this out, or push forward to create the site for mapi?

**Comment by JoelSpeed** [comment]
Let's close this until we have further time to push it forward

**Pull request closed without merging**